classes/image_populate_mfgtool.bbclass

```diff
@@ -0,0 +1,142 @@
+# separated by commas. You can omit any argument you like but must retain the separating commas. The
+# order is important and specifies the following:
+#
+#   1. Extra dependencies that should be added to the do_populate_mfgtool task, if the script is
+#      enabled.
+#   2. Extra binaries that should be added to the bundle, if the script is enabled.
+#
+# For example:
+#
+# ,----[ Defining bootloader.uuu.in script ]
+# | MFGTOOL_SCRIPT = "bootloader.uuu.in"
+# | MFGTOOL_SCRIPT[bootloader.uuu.in] = "dep1:do_deploy dep2:do_deploy,file1 file2"
+# `----
+#
+# The virtual/bootloader:do_deploy is added to do_populate_mfgtool dependencies and ${UBOOT_BINARY}
+# copied to the bundle, only if the script is enabled.
+#
+# During the mfgtool bundle generation, the uuu.in files are processed and some variables
+# replaced. The variables are:
+#
+#   - MACHINE
+#   - UBOOT_BINARY
+#   - SPL_BINARY
+#   - IMAGE_BASENAME
+#
+# Copyright 2022-2023 (C) O.S. Systems Software LTDA.
+#
+# SPDX-License-Identifier: MIT
+
+MFGTOOL_FILESPATH ??= " \
+    ${@base_set_filespath(["%s/mfgtool" % p for p in "${BBPATH}".split(":")] \
+                             + ["${FILE_DIRNAME}/${BP}/mfgtool", \
+                                "${FILE_DIRNAME}/${BPN}/mfgtool", \
+                                "${FILE_DIRNAME}/files/mfgtool"] \
+                          , d)} \
+"
+
+MFGTOOLDIR = "${WORKDIR}/mfgtool-${PN}"
+do_populate_mfgtool[dirs] = "${MFGTOOLDIR}"
+do_populate_mfgtool[cleandirs] = "${MFGTOOLDIR}"
+
+addtask populate_mfgtool after do_image_complete do_unpack before do_deploy
+do_populate_mfgtool[dirs] ?= "${DEPLOY_DIR_IMAGE} ${WORKDIR}"
+do_populate_mfgtool[nostamp] = "1"
+do_populate_mfgtool[recrdeptask] += "do_deploy"
+do_populate_mfgtool[depends] += "uuu-bin-native:do_populate_sysroot"
+
+python () {
+    # Process MFGTOOL_SCRIPT and handle its fields.
+    script_flags = d.getVarFlags('MFGTOOL_SCRIPT') or {}
+    script_depends = []
+    script_deploy_files = []
+    if script_flags:
+        script = (d.getVar('MFGTOOL_SCRIPT') or "").split()
+        pn = d.getVar("PN")
+
+        for flag, flagval in sorted(script_flags.items()):
+            items = flagval.split(",")
+            num = len(items)
+            if num > 2:
+                bb.error("%s: MFGTOOL_SCRIPT[%s] Only \"depends,deploy files\" can be specified!" % (pn, flag))
+
+            if flag in script:
+                if num >= 2 and items[1]:
+                    script_deploy_files.append(items[1])
+                if num >= 1 and items[0]:
+                    script_depends.append(items[0])
+
+        d.appendVarFlag('do_populate_mfgtool', 'depends', ' ' + ' '.join(script_depends))
+
+    d.setVar('_SCRIPT_DEPLOY_FILES', ' '.join(script_deploy_files))
+}
+
+python do_populate_mfgtool() {
+    # For MFGTOOL_SCRIPT items we use BitBake's fetcher module allowing a consistent behavior.
+    script = (d.getVar('MFGTOOL_SCRIPT') or "").split()
```

Comment:
Consider if you wanted to do a `for-loop`, best would be `for script in scripts`.

Reply:
Sure, but the `MFGTOOL_SCRIPT` is analogous to `PACKAGECONFIG`. It defines a script to have flags (depends and files) defined.

Reply:
Let me try again: I consider that `PACKAGECONFIG` has a special definition, that a configuration is a set of values defining the singular configuration. I see no such definition for `MFGTOOL_SCRIPT`. It is not a set of values defining a singular script, so it is a bad analog.

Reply:
Hmm, how about using `MFGTOOLCONFIG`? I guess that makes your analog clearer to me.

Reply:
`MFGTOOL_SCRIPT`: I consider it defines a script with a set of values. Consider this:

```
MFGTOOL_SCRIPT = "foo.uuu.in bar.uuu"
MFGTOOL_SCRIPT[foo.uuu.in] = "dep-foo1:do_deploy dep-foo2:do_deploy,file-foo1 file-foo2"
MFGTOOL_SCRIPT[bar.uuu] = "dep-bar1:do_deploy dep-bar2:do_deploy,file-bar1 file-bar2"
```

This defines two scripts, and each has a specific set of values. A convention is to use the same variable (here `MFGTOOL_SCRIPT`) to look up the flags.

Reply:
The question is whether the variable name can be improved. Calling it `MFGTOOL_SCRIPT` gives a natural connotation that only one script is allowed. A knowing user will go to the docs to see if multiples are allowed. Making it plural makes it clear from the name.

As I said, I think all this goes away if you just make the variable `MFGTOOL_CONFIG`, aligning it with `PACKAGECONFIG` and `UBOOT_CONFIG`, so giving the user another clue about how it works just from the name.

Reply:
Fixed.